core/src/crypto.rs

```
@@ -0,0 +1,20 @@
+use crate::prelude::*;
+use crate::Address;
+use casper_types::account::AccountHash;
+use casper_types::{PublicKey, SecretKey};
+
+pub fn generate_key_pairs(amount: u8) -> BTreeMap<Address, (SecretKey, PublicKey)> {
+    let mut accounts = BTreeMap::new();
+    for i in 0..amount {
+        // Create keypair.
+        let secret_key = SecretKey::ed25519_from_bytes([i; 32]).unwrap();
+        let public_key = PublicKey::from(&secret_key);
+
+        // Create an AccountHash from a public key.
+        let account_addr = AccountHash::from(&public_key);
+
+        // Create a GenesisAccount.
+        accounts.insert(account_addr.try_into().unwrap(), (secret_key, public_key));
```

As @zie1ony stated https://github.com/odradev/odra/pull/272#discussion_r1413704292 core functions should not fail (except reverts)

---

Well, if account hash cannot be converted into address, something is terribly wrong. Nevertheless, I pushed a version which continues in this case.

---

I'm not sure if skipping is the best strategy, we just swallow an error. Maybe we should panic or revert depending on the context. A few lines above `let secret_key = SecretKey::ed25519_from_bytes([i; 32]).unwrap()` you unwrap a SecretKey creation. We could handle it better, but agree, very likely it'll never fail. I'm not insisting, just sayin'.

---

Now it panics :)

---

If used in tests only then fine. :)